chirp_client.cc

```diff
@@ -0,0 +1,302 @@
+    std::cout << "  [" << reply_chirps.at(0).username() << "]" << std::endl;
+    std::cout << "  ID: " << reply_chirps.at(0).id(); 
+    if(!reply_chirps.at(0).parent_id().empty()) {
+        std::cout << ", (reply to: " << reply_chirps.at(0).parent_id() << ")";
+    }
+    std::cout << std::endl;
+    std::cout << "  - '" << reply_chirps.at(0).text()  << "'" << std::endl;
+    std::cout << "  " << google::protobuf::util::TimeUtil::ToString(google::protobuf::util::TimeUtil::SecondsToTimestamp(reply_chirps.at(0).timestamp().seconds())) << std::endl;
+    std::cout << "|----------------------------------|" << std::endl;
+  }
+
+  // Is a thread so we need to tab correctly
+  else {
+    std::string tab = "\t";
+    std::string total_tabs = "";
+    for(chirp::Chirp chirp : reply_chirps) {
+      total_tabs = "";
+
+      for(int i = 1; i < chirp.depth(); ++i) {
+        total_tabs += tab;
+      }
+      std::cout << total_tabs << "|----------------------------------|" << std::endl;
+      std::cout << total_tabs << "  [" << chirp.username() << "]" << std::endl;
+      std::cout << total_tabs <<"  ID: " << chirp.id();
+      if(!chirp.parent_id().empty()) {
+        std::cout << ", (reply to ID: " << chirp.parent_id() << ")";
+      }
+      std::cout << std::endl;
+      std::cout << total_tabs << "  - '" << chirp.text() << "'" << std::endl;
+      std::cout << total_tabs << "  " << google::protobuf::util::TimeUtil::ToString(google::protobuf::util::TimeUtil::SecondsToTimestamp(chirp.timestamp().seconds())) << std::endl;
+      std::cout << total_tabs << "|----------------------------------|" << std::endl;
+    }
+  }
+
+}
+// Instantiate client. It requires a channel, out of which the actual RPCs
+// are created. This channel models a connection to an endpoint (in this case,
+// localhost at port 50000). We indicate that the channel isn't authenticated
+// (use of InsecureChannelCredentials()).
+int main(int argc, char** argv) {
+  ChirpClient greeter(grpc::CreateChannel("localhost:50000", grpc::InsecureChannelCredentials()));
+  google::InitGoogleLogging(argv[0]);
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+  std::string register_user = FLAGS_register;
+  std::string user = FLAGS_user;
+  std::string chirp = FLAGS_chirp;
+  std::string parent_id = FLAGS_reply;
+  std::string read_id = FLAGS_read;
+  std::string follow = FLAGS_follow;
+  bool monitor = FLAGS_monitor;
+
+  if(!register_user.empty()) {
+    greeter.registeruser(register_user);
+    return 0;
+  }
+  if(!read_id.empty())
+  {
+    greeter.read(read_id);
+    return 0;
+  }
+  if(!parent_id.empty() && chirp.empty()) {
+    std::cout << "Must specific chirp ID to reply to " << std::endl;
+  }
+  if(!chirp.empty() && !user.empty())
+  {
+    greeter.chirp(user, chirp, parent_id);
+    return 0;
+  }
+  if(!follow.empty() && !user.empty()) {
+     greeter.follow(user, follow);
+     return 0;
+  }
+  if(monitor && !user.empty())
```

Comment:
!user.empty() is redundant in  lines 284-293—you could surround all the logic in an if statement checking user.empty(). (at least that's what I was told to do for one PR)